### PR TITLE
feat: add edit name for plan

### PR DESCRIPTION
**Pull request by lionel95200x**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/savings-plan-epic` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #TAPC-856
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Add edit name on a Savings Plan

### CODE REVIEW

File: packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          e: OsdsInputCustomEvent<OdsInputValueChangeEventDetail>,
+        ) => setNewName(e.target.value as string)}
+      />
+      <OsdsButton
+        slot="actions"
+        variant={ODS_BUTTON_VARIANT.stroked}
+        color={ODS_THEME_COLOR_INTENT.primary}
+        onClick={() => {
+          onClose();
+        }}
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx:61]:
```suggestion
        onClick={onClose}
```

File: packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        color={
+          isValidName
+            ? ODS_THEME_COLOR_INTENT.info
+            : ODS_THEME_COLOR_INTENT.error
+        }
+        className="p-3"
+        value={newName}
+        onOdsValueChange={(
+          e: OsdsInputCustomEvent<OdsInputValueChangeEventDetail>,
+        ) => setNewName(e.target.value as string)}
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx:53]: We could avoid binding lambda functions in JSX by implementing a `handleOdsValueChange` method

File: packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      >
+        {t('cancel')}
+      </OsdsButton>
+      <OsdsButton
+        disabled={!isButtonValid || undefined}
+        slot="actions"
+        color={ODS_THEME_COLOR_INTENT.primary}
+        onClick={() => {
+          onConfirm(newName);
+        }}
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx:71]: Same here, as a JSX improvement
```suggestion
        onClick={handleConfirm(newName}
```

File: packages/manager/apps/pci-savings-plan/src/components/Table/ActionsCell.tsx [hunk 4]
```diff
@@ -77,7 +77,7 @@ export default function ActionsCell({
           size={ODS_BUTTON_SIZE.sm}
           variant={ODS_BUTTON_VARIANT.ghost}
           text-align="start"
-          onClick={() => onClickManage(id)}
+          onClick={() => onClickRenew()}
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Table/ActionsCell.tsx:80]:
```suggestion
          onClick={onClickRenew}
```

File: packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+export default function EditNameModal({
+  oldName,
+  onClose,
+  onConfirm,
+}: Readonly<TEditNameModal>) {
+  const { t } = useTranslation('edit-name');
+  const [newName, setNewName] = useState(oldName);
+
+  const isValidName = /^[a-z0-9][-_.A-Za-z0-9]{1,61}[a-z0-9]$/.test(newName);
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx:31]: Maybe this one could be a dedicated utility function with a dedicated test suite, what do you think?

File: packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        color={ODS_THEME_COLOR_INTENT.primary}
+        onClick={onClose}
+      >
+        {t('cancel')}
+      </OsdsButton>
+      <OsdsButton
+        disabled={!isButtonValid || undefined}
+        slot="actions"
+        color={ODS_THEME_COLOR_INTENT.primary}
+        onClick={() => onConfirm(newName)}
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Modal/EditNameModal.tsx:70]: I think here that we can harmonize the use of handler functions by removing all anonymous lambdas that are systematically recreated during the render phase and might lead to performance issues.
For this, in order to manage the parameter, we can use function composition and a bit of curry :)

```suggestion
        const handleConfirm = (name: string) => () => onConfirm(name);
        onClick={handleConfirm(newName)}
```

File: packages/manager/apps/pci-savings-plan/src/components/Table/ActionsCell.tsx [hunk 7]
```diff
@@ -62,7 +62,7 @@ export default function ActionsCell({
             size={ODS_BUTTON_SIZE.sm}
             variant={ODS_BUTTON_VARIANT.ghost}
             text-align="start"
-            onClick={() => onClickManage(id)}
+            onClick={() => onClickEditName(id)}
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/components/Table/ActionsCell.tsx:65]: Same here, we can improve this with currying :)

File: packages/manager/apps/pci-savings-plan/src/pages/listing/renew-modal/index.tsx [hunk 8]
```diff
@@ -1,52 +1,42 @@
...
+      onClose={() => navigate('..')}
+      onConfirm={() => {
+        changePeriod({
+          periodEndAction: periodEndAction
+            ? SavingsPlanPlanedChangeStatus.TERMINATE
+            : SavingsPlanPlanedChangeStatus.REACTIVATE,
+        });
+        navigate('..');
+      }}
+    />
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-savings-plan/src/pages/listing/renew-modal/index.tsx:37]: Same here :)

**Review by fredericvilcot — COMMENTED** [review]
Almost there :)

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Review by fredericvilcot — APPROVED** [review]
LGTM, thanks for the changes :)

**Pull request merged**